pkg/kubelet/runtimeregistry/kuberuntime_registry.go

```diff
@@ -0,0 +1,295 @@
+	GetAllRuntimeServices() (map[string]*RuntimeService, error)
+	// Get all image services supported on the node
+	GetAllImageServices() ( map[string]*ImageService, error)
+	// Get the primary runtime service for the Arktos cluster
+	GetPrimaryRuntimeService() (*RuntimeService, error)
+	// Get the runtime service for particular workload type(VM or container)
+	GetRuntimeServiceByWorkloadType(workloadtype string) (*RuntimeService, error)
+	// Get the runtime service for particular workload type(VM or container)
+	GetImageServiceByWorkloadType(workloadtype string) (*ImageService, error)
+	// Get status for all runtime services
+	GetAllRuntimeStatus() (map[string]map[string]bool, error)
+
+}
+
+func NewKubeRuntimeRegistry (remoteRuntimeEndpoints string) (*KubeRuntimeRegistry, error) {
+
+	rs, is, err := buildRuntimeServicesMapFromAgentCommandArgs(remoteRuntimeEndpoints)
+
+	if err != nil {
+		klog.Errorf("Failed create the runtime service maps. Error: %v", err)
+		return nil, err
+	}
+
+	return &KubeRuntimeRegistry{RuntimeServices: rs, ImageServices: is}, nil
+}
+
+func (r * KubeRuntimeRegistry) GetAllRuntimeServices() (map[string]*RuntimeService, error) {
+	return r.RuntimeServices, nil
+}
+
+func (r *KubeRuntimeRegistry) GetAllImageServices() ( map[string]*ImageService, error) {
+	return r.ImageServices, nil
+}
+
+func (r *KubeRuntimeRegistry) GetPrimaryRuntimeService() (*RuntimeService, error) {
+	for _, runtimeService := range r.RuntimeServices {
+		if runtimeService.IsPrimary == true {
+			return runtimeService, nil
+		}
+	}
+
+	return nil, fmt.Errorf("primary runtime servcie is not defined")
+}
+
+func (r *KubeRuntimeRegistry) GetRuntimeServiceByWorkloadType(workloadtype string) (*RuntimeService, error) {
+	for _, runtimeService := range r.RuntimeServices {
+		if runtimeService.WorkloadType == workloadtype {
+			return runtimeService, nil
+		}
+	}
+
+	return nil, fmt.Errorf("runtime servcie for workload type %v is not defined", workloadtype)
+}
+
+func (r *KubeRuntimeRegistry) GetImageServiceByWorkloadType(workloadtype string) (*ImageService, error) {
+	for _, imageService := range r.ImageServices {
+		if imageService.WorkloadType == workloadtype {
+			return imageService, nil
+		}
+	}
+
+	return nil, fmt.Errorf("image servcie for workload type %v is not defined", workloadtype)
+}
+
+// Get status for all runtime services
+func (r *KubeRuntimeRegistry) GetAllRuntimeStatus() (map[string]map[string]bool, error) {
+	statuses := make(map[string]map[string]bool)
+	vmServices := make(map[string]bool)
+	containerServices := make(map[string]bool)
+
+	for runtimeName, runtimeService := range r.RuntimeServices {
+		workloadType := runtimeService.WorkloadType
+		runtimeReady := true
+
+		status, err := runtimeService.ServiceApi.Status()
+		if err != nil || status == nil {
+			runtimeReady = false
+		}
+
+		for _, c := range status.GetConditions() {
+			if c.Status != true {
+				runtimeReady = false
+				break
+			}
+		}
+
+		if workloadType == "vm" {
```

Comment:
make this "vm" a constant

Reply:
this is a miss when the two constances were added. just fixed. thanks.

Reply:
committed fixes.